src/cli/abstraction/mod.rs

```diff
@@ -0,0 +1,32 @@
+use clap::Args;
+
+#[derive(Args, Debug)]
+pub(crate) struct AuthArguments {
+	#[arg(help = "Database authentication username to use when connecting")]
+	#[arg(env = "SURREAL_USER", short = 'u', long = "username", visible_alias = "user")]
+	#[arg(default_value = "root")]
+	pub(crate) username: String,
+	#[arg(help = "Database authentication password to use when connecting")]
+	#[arg(short = 'p', long = "password", visible_alias = "pass")]
+	#[arg(env = "SURREAL_PASS", default_value = "root")]
+	pub(crate) password: String,
+}
+
+#[derive(Args, Debug)]
+pub struct DatabaseSelectionArguments {
+	#[arg(help = "The namespace selected for the operation")]
+	#[arg(env = "SURREAL_NAMESPACE", long = "namespace", visible_alias = "ns")]
+	pub(crate) namespace: String,
+	#[arg(help = "The database selected for the operation")]
+	#[arg(env = "SURREAL_DATABASE", long = "database", visible_alias = "db")]
+	pub(crate) database: String,
+}
+
+#[derive(Args, Debug)]
+pub struct DatabaseConnectionArguments {
+	#[arg(help = "Remote database server url to connect to")]
+	#[arg(short = 'c', long = "connection-url", visible_aliases = ["host", "conn"])]
```

Comment:
@tobiemh Should we take this opportunity to rename this to `endpoint` or something like that? Since beta 9, this now accepts

- `memory` or `mem://`
- `ws://localhost:8000` or `wss://localhost:8000`
- `http://localhost:8000` or `https://localhost:8000`
- `indxdb://DbName`
- `file:///path/to/db` or `rocksdb:///path/to/db`
- `tikv://localhost:2379`
- `fdb:///path/to/fdb.cluster`

On the API layer we use endpoint to refer to these.

Reply:
That's probably a good idea @rushmorem yes. @finnbear 👀

Reply:
Cool, I've changed it to `--endpoint` with the previous `--conn` as a visible alias.